### ISSUE TITLE
📣 You can respond to PE bug reports now

### ISSUE THREAD

**Issue by damithc** [nus-cs2103-AY1718S1/forum#233]
Bugs others reported for your app are now available on [pe-results issue tracker](https://github.com/nus-cs2103-AY1718S1/pe-results/issues). You can filter by labels to find issues related to your team. 
![image](https://user-images.githubusercontent.com/1673303/32991705-aa8e9f72-cd7b-11e7-895a-1e78b6ac910b.png)

We are going to give you a few days (until **Wednesday 22nd Nov midnight**) to respond to those bug reports before we start grading them. Here are the instructions. This process is recommended to be done as a team.

* **First, don't freak out if there are lot of bug reports.** Many can be duplicates and some can be _false positives_. In any case, we anticipate that all of these products will have some bugs and our penalty for bugs is not harsh. Furthermore, it depends on the severity of the bug. Some bug may not even be penalized.

* **Do not edit the subject or the description. Do not close bug reports.** Your response (if any) should be added as a comment.

* **If the bug is reported multiple times**, mark all copies EXCEPT one as duplicates using the `duplicate` tag. In addition, use [this technique](https://help.github.com/articles/about-duplicate-issues-and-pull-requests/) to indicate which issue they are duplicates of. Duplicates can be omitted from processing steps given below.

* **If a bug seems to be for a different product** (i.e. wrongly assigned to your team), let us know (email prof).

* **Decide if it is a real bug and apply ONLY one of these labels**.
  * `response.Accepted` : You accept it as a bug. 
  * `response.Rejected` : What tester thought as a bug is in fact expected behavior. :exclamation: The penalty for rejecting a bug using an unjustifiable explanation is higher than the penalty if the same bug was accepted.
  * `response.CannotReproduce` : You are unable to reproduce the behavior reported in the bug after multiple tries. 
  * `response.IssueUnclear` : The issue description is not clear.

* If applicable, **decide the type of bug**:
  * `type-FunctionalityBug` : the bug is a flaw in how the product works
  * `type-DocumentationBug` : the bug is in the documentation

* If a functionality bug, **suggest a severity for the bug**:
  * `severity.Low` : Unlikely to affect normal operations of the product. Appears only in very rare situations and causes a minor inconvenience only.
  * `severity.Medium` : Causes occasional inconvenience to some users but they can continue to use the product.
  * `severity.High` : Affects most users and causes major problems for users. i.e., makes the product almost unusable for most users.

* **Decide who should fix the bug**. Use the `Assignees` field to assign the issue to that person(s). There is no need to actually fix the bug though. It's simply an indication/acceptance of responsibility. **If there is no assignee and we are unable to determine a suitable assignee ourselves, we'll have no choice but to distribute the penalty for that bug (if any) among all team members.**

* **Add an explanatory comment** explaining your choice of labels and assignees.

**Comment by erik0704** [comment]
Hi professor,
Correct me if am I wrong but I think we cannot apply labels (and assignee) to the issue if we are not a contributor ?

**Comment by damithc** [comment]
> Hi professor,
> Correct me if am I wrong but I think we cannot apply labels (and assignee) to the issue if we are not a contributor ?

@erik0704 Had the wrong settings before. Should be able to add labels now.

**Comment by Adoby7** [comment]
Hi Prof,
If there is no detailed information/ reproduce steps/ screenshots, can we simply reject it?
See this example:
![picture](https://user-images.githubusercontent.com/25312978/32991995-59514380-cd80-11e7-9bb2-15832fd7736f.png)

**Comment by teclu** [comment]
Hi Prof, some of the issues were wrongly assigned to my group. What do we do?

**Comment by damithc** [comment]
> Hi Prof, some of the issues were wrongly assigned to my group. What do we do?

Email me which issues are wrongly assigned

**Comment by damithc** [comment]
> If there is no detailed information/ reproduce steps/ screenshots, can we simply reject it?

Yes

**Comment by archthegit** [comment]
Hi Prof, what do we do about existing bugs when the application was first given to us? for e.g.: after using the find command and then deleting a person, and then returning to the normal list and undoing and redoing, the same person doesn't get deleted.

**Comment by damithc** [comment]
> Hi Prof, what do we do about existing bugs when the application was first given to us? for e.g.: after using the find command and then deleting a person, and then returning to the normal list and undoing and redoing, the same person doesn't get deleted.

You can reject them but first report it in [se-edu/addressbook-level4](https://github.com/se-edu/addressbook-level4/issues) to get it confirmed as an existing bug.

**Comment by philemontan** [comment]
Hi Prof,
The team.B1 and team.B2 labels seem to have been removed across all issues by a single student. What should we do?

**Comment by damithc** [comment]
> The team.B1 and team.B2 labels seem to have been removed across all issues by a single student. What should we do?

@philemontan they still missing? Someone removed some labels by mistake but added them back later.

**Comment by Hailinx** [comment]
Hi Prof,
What if the issue reported is the improvement? We never mention our todo cannot add time which had passed.
![image](https://user-images.githubusercontent.com/28561970/32999315-3ac7b514-cddc-11e7-8355-8406110ab8a7.png)

**Comment by damithc** [comment]
> Hi Prof,
> What if the issue reported is the improvement? We never mention our todo cannot add time which had passed.

@Hailinx can reject those

**Comment by yunpengn** [comment]
Hi prof! If one command is not undoable now, but the tester suggests that it is more reasonable to make it undoable.

We never specify the command is undoable, but I agree with the tester it would be a better UX if it is undoable. And I certainly would do like this if there is v2.0
<br>
Is this considered to be an improvement or a bug?

**Comment by damithc** [comment]
> We never specify the command is undoable, but I agree with the tester it would be a better UX if it is undoable. And I certainly would do like this if there is v2.0

@yunpengn can consider it as a suggestion.

**Comment by archthegit** [comment]
Hi Prof, if a bug was rejected because it was a bug with the original addressbook, do we still need to assign it to someone?

**Comment by damithc** [comment]
> Hi Prof, if a bug was rejected because it was a bug with the original addressbook, do we still need to assign it to someone?

@archthegit No.

**Comment by tshradheya** [comment]
Prof @damithc , what if there is an issue which states that the command takes long to execute(1 s) but that is the expected behavior since some operations like reading and writing image have to be done. Is this more like a suggestion or a bug?

**Comment by damithc** [comment]
> Prof @damithc , what if there is an issue which states that the command takes long to execute(1 s) but that is the expected behavior since some operations like reading and writing image have to be done. Is this more like a suggestion or a bug?

Not a bug in the scope of this PE. can be rejected.

**Comment by yunpengn** [comment]
Hi prof! We are using Natty to do natural language passing, but a bug report found that Natty returns the wrong result.

The input is invalid but Natty returns a weird result without any error. It seems that Natty suspects some typo in the input string and tries to correct the typo _automatically_ (but unfortunately, the correction is wrong).

I know it comes with an expense when using external library. But natural language passing is still a research topic, we cannot really do much to `fix` this bug.

We also stated in the user guide that
> the application may sometimes be smart enough to interpret what you typed. For example, if you type this afternoon or tomorrow evening, it will be automatically converted to the standard format.

> Do NOT try to challenge the application, it is not promised to produce an expected result if you use non-standard format or if the expression is not simple or clear enough.

Should we accept this bug or reject it?

**Comment by damithc** [comment]
> Hi prof! We are using Natty to do natural language passing, but a bug report found that Natty returns the wrong result.
> 
> The input is invalid but Natty returns a weird result without any error. It seems that Natty suspects some typo in the input string and tries to correct the typo automatically (but unfortunately, the correction is wrong).
> 
> I know it comes with an expense when using external library. But natural language passing is still a research topic, we cannot really do much to fix this bug.
> 
> We also stated in the user guide that
> 
> the application may sometimes be smart enough to interpret what you typed. For example, if you type this afternoon or tomorrow evening, it will be automatically converted to the standard format.
> Do NOT try to challenge the application, it is not promised to produce an expected result if you use non-standard format or if the expression is not simple or clear enough.
> Should we accept this bug or reject it?

Yes you can reject it. From what I remember, Natty will interpret Feb 29th (if not a leap year) as March 1st and so on.

**Comment by Hailinx** [comment]
Hi Prof,
Java API LocalDataTime has the same problem. It will interpret Feb 29th in non leap year to Feb 28th. And an issue reports on this. Should I reject the issue?

**Comment by damithc** [comment]
> Java API LocalDataTime has the same problem. It will interpret Feb 29th in non leap year to Feb 28th. And an issue reports on this. Should I reject the issue?

Yup. Can reject.
